apps/webhooks/lib/Db/WebhookListenerMapper.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Webhooks\Db;
+
+use OCP\AppFramework\Db\DoesNotExistException;
+use OCP\AppFramework\Db\MultipleObjectsReturnedException;
+use OCP\AppFramework\Db\QBMapper;
+use OCP\DB\Exception;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\EventDispatcher\IWebhookCompatibleEvent;
+use OCP\ICache;
+use OCP\ICacheFactory;
+use OCP\IDBConnection;
+
+/**
+ * @template-extends QBMapper<WebhookListener>
+ */
+class WebhookListenerMapper extends QBMapper {
+	public const TABLE_NAME = 'webhook_listeners';
+
+	private const EVENTS_CACHE_KEY = 'eventsUsedInWebhooks';
+
+	private ?ICache $cache = null;
+
+	public function __construct(
+		IDBConnection $db,
+		ICacheFactory $cacheFactory,
+	) {
+		parent::__construct($db, self::TABLE_NAME, WebhookListener::class);
+		if ($cacheFactory->isAvailable()) {
+			$this->cache = $cacheFactory->createDistributed();
+		}
+	}
+
+	/**
+	 * @throws DoesNotExistException
+	 * @throws MultipleObjectsReturnedException
+	 * @throws Exception
+	 */
+	public function getById(int $id): WebhookListener {
+		$qb = $this->db->getQueryBuilder();
+
+		$qb->select('*')
+			->from($this->getTableName())
+			->where($qb->expr()->eq('id', $qb->createNamedParameter($id, IQueryBuilder::PARAM_INT)));
+
+		return $this->findEntity($qb);
+	}
+
+	/**
+	 * @throws Exception
+	 * @return WebhookListener[]
+	 */
+	public function getAll(): array {
+		$qb = $this->db->getQueryBuilder();
+
+		$qb->select('*')
+			->from($this->getTableName());
+
+		return $this->findEntities($qb);
+	}
+
+	/**
+	 * @throws Exception
+	 */
+	public function addWebhookListener(
+		?string $appId,
+		string $userId,
+		string $httpMethod,
+		string $uri,
+		string $event,
+		?array $eventFilter,
+		?array $headers,
+		AuthMethod $authMethod,
+		?array $authData,
```

Comment:
Should we start adding:
```suggestion
		#[\SensitiveParameter]
		?array $authData,
```
for new things at least?